libkbfs/journal_md_ops.go

```diff
@@ -183,6 +217,12 @@ func (j journalMDOps) getRangeFromJournal(
 	irmds := make([]ImmutableRootMetadata, 0, len(ibrmds))
 
 	for _, ibrmd := range ibrmds {
+		if bid == LocalSquashBranchID && ibrmd.BID() != bid {
```

Comment:
I'll return an error if `ibrmd.BID() != NullBranchID`, but I'm not sure what the check on line 201 has to do with this loop.  That is only checking the head, and since that check has passed, these checks only apply to earlier MDs.